api/src/main/java/org/apache/iceberg/DeleteFiles.java

```diff
@@ -51,10 +51,7 @@
    * @param file a DataFile to remove from the table
    * @return this for method chaining
    */
-  default DeleteFiles deleteFile(DataFile file) {
```

Comment:
I don't think it is necessary to remove the default implementation. Overriding it in `StreamingDelete` should work just fine without causing a binary compatibility change.